shared/Microsoft.Extensions.ActivatorUtilities.Sources/ActivatorUtilities.cs

```diff
@@ -0,0 +1,305 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Runtime.ExceptionServices;
+
+namespace Microsoft.Extensions.Internal
+{
+    internal delegate object ObjectFactory(IServiceProvider serviceProvider, object[] arguments);
+
+    // Do not take a dependency on this class unless you are explicitly trying to avoid taking a
```

Comment:
This comment is not super useful. I don't think anyone actually reads these messages.